### PR TITLE
Add drone support.

### PR DESCRIPTION
**Pull request by DroneEnthusiast**

### CODE REVIEW

**Comment by Spice-King** [comment]
Don't know why GitHub closed this on me

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "build:watch": "gulp link watch",
     "clean": "gulp clean && gulp link --clean",
     "foundry": "node foundry/resources/app/main.js --noupnp --dataPath=./data",
-    "update": "npm install --save-dev gitlab:spice_king/foundry-pc-types"
+    "update": "npm install --save-dev gitlab:a.drone.enthusiast/foundry-pc-types"
```
Comment (Spice-King) [inline package.json:12]: Don't change deps for no good reason in a random commit.

File: src/module/ValidatedDialog.ts [hunk 2]
```diff
@@ -22,14 +22,13 @@ export class ValidatedDialog extends Dialog {
     super(dialogData, options);
     this.failCallback = dialogData.failCallback;
   }
-  validate(button: ButtonData): boolean {
+  validate(): boolean {
```
Comment (Spice-King) [inline src/module/ValidatedDialog.ts:25]: Save random clean ups for another PR

File: src/module/actors/character-sheet.ts [hunk 3]
```diff
@@ -247,7 +248,7 @@ export class CharacterActorSheet extends ActorSheet<
 
     this.popUpDialog = new ValidatedDialog(
       {
-        failCallback: (button: ButtonData): void => {
+        failCallback: (): void => {
```
Comment (Spice-King) [inline src/module/actors/character-sheet.ts:251]: Again, no random changes outside the idea of the PR.

File: src/module/actors/npc.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ export class SWNRNPCActor extends Actor<SWNRNPCData> {
     userId: string
   ): void {
     super._onCreate(data, options, userId);
-    if ((this.data as any).items.length || game.userId !== userId) return;
+    if (this.data["items"]["length"] || game.userId !== userId) return;
```
Comment (Spice-King) [inline src/module/actors/npc.ts:15]: Again no random unrelated reformatting.

File: src/module/handlebar-helpers.ts [hunk 5]
```diff
@@ -5,12 +5,48 @@ export default function registerHelpers(): void {
...
-    return JSON.stringify(obj, null, 2);
-  });
-  Handlebars.registerHelper("concat", function (a, b) {
+  Handlebars.registerHelper(
+    "stringify",
+    function (obj: Record<string, unknown>) {
+      return JSON.stringify(obj, null, 2);
+    }
+  );
+  Handlebars.registerHelper("concat", function (a: string, b: string) {
```
Comment (Spice-King) [inline src/module/handlebar-helpers.ts:14]: No random changes outside the PR idea

File: src/module/items/base-sheet.ts [hunk 6]
```diff
@@ -1,3 +1,8 @@
+interface EditorOptions {
+  target: Record<string, unknown>;
+  height: number;
+  save_onsavecallback: () => Promise<void>;
+}
```
Comment (Spice-King) [inline src/module/items/base-sheet.ts:5]: Once again, off topic for the PR's stated goal

File: src/module/types.d.ts [hunk 7]
```diff
@@ -66,6 +68,31 @@ declare interface SWNRCharacterData extends SWNRLiving, SWNREncumbrance {
...
+  model:
+    | "primitiveDrone"
+    | "voidHawk"
+    | "stalker"
+    | "cuttlefish"
+    | "ghostwalker"
+    | "sleeper"
+    | "pax"
+    | "alecto"
+    | "other";
```
Comment (Spice-King) [inline src/module/types.d.ts:81]: This limits extendability. Should just be a plain string.
```suggestion
  model: string;
```

File: src/module/actors/character.ts [hunk 8]
```diff
@@ -9,8 +9,8 @@ import {
 import { SWNRBaseItem } from "../base-item";
 
 export class SWNRCharacterActor extends Actor<SWNRCharacterData> {
-  getRollData(): Record<string, any> {
-    const data = super.getRollData() as Record<string, any>;
+  getRollData(): Record<string, unknown> {
+    const data = super.getRollData() as Record<string, unknown>;
```
Comment (Spice-King) [inline src/module/actors/character.ts:13]: Once again, random clean ups are for a dedicated PR.

File: src/module/actors/drone-sheet.ts [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+                        encumbrance: 4,
+                        range: 5000,
+                        tl: 5,
+                        description: game.i18n.localize(
+                          "swnr.drone.model.alecto.text"
+                        ),
+                      },
+                    });
+                    break;
+                }
```
Comment (Spice-King) [inline src/module/actors/drone-sheet.ts:217]: This whole switch statement is a code smell. It contains data that should be totally external and blocks extension by users or other modules.

I've been tossing around the idea of a hull item for Ships which spec out the base details of a hull, though including an actor compendium for drones is probably a simpler method.

File: src/packs/heavy/Anti-Vehicle Laser.yml [hunk 10]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: special
     value: 15
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/heavy/Anti-Vehicle Laser.yml:13]: Don't remove data unrelated to PR

File: src/packs/heavy/Demo Charge.yml [hunk 11]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: special
     value: 0
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/heavy/Demo Charge.yml:13]: Don't remove data unrelated to PR

File: src/packs/heavy/Heavy Machine Gun.yml [hunk 12]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: true
     type: special
     value: 10
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/heavy/Heavy Machine Gun.yml:13]: Don't remove data unrelated to PR

File: src/packs/heavy/Hydra Array.yml [hunk 13]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: true
     type: special
     value: 10
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/heavy/Hydra Array.yml:13]: Don't remove data unrelated to PR

File: src/packs/heavy/Railgun.yml [hunk 14]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: true
     type: special
     value: 20
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/heavy/Railgun.yml:13]: Don't remove data unrelated to PR

File: src/packs/heavy/Rocket Launcher.yml [hunk 15]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: special
     value: 1
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/heavy/Rocket Launcher.yml:13]: Don't remove data unrelated to PR

File: src/packs/heavy/Vortex Cannon.yml [hunk 16]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: special
     value: 5
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/heavy/Vortex Cannon.yml:13]: Don't remove data unrelated to PR

File: src/packs/heavy/Wheatcutter Belt.yml [hunk 17]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: special
     value: 5
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/heavy/Wheatcutter Belt.yml:13]: Don't remove data unrelated to PR

File: src/packs/melee/Large advanced weapon.yml [hunk 18]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: none
     value: 0
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/melee/Large advanced weapon.yml:13]: Don't remove data unrelated to PR

File: src/packs/melee/Large primitive weapon.yml [hunk 19]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: none
     value: 0
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/melee/Large primitive weapon.yml:13]: Don't remove data unrelated to PR

File: src/packs/melee/Medium advanced weapon.yml [hunk 20]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: none
     value: 0
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/melee/Medium advanced weapon.yml:13]: Don't remove data unrelated to PR

File: src/packs/melee/Medium primitive weapon.yml [hunk 21]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: none
     value: 0
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/melee/Medium primitive weapon.yml:13]: Don't remove data unrelated to PR

File: src/packs/melee/Small advanced weapon.yml [hunk 22]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: none
     value: 0
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/melee/Small advanced weapon.yml:13]: Don't remove data unrelated to PR

File: src/packs/melee/Small primitive weapon.yml [hunk 23]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: none
     value: 0
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/melee/Small primitive weapon.yml:13]: Don't remove data unrelated to PR

File: src/packs/melee/Stun baton.yml [hunk 24]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: none
     value: 0
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/melee/Stun baton.yml:13]: Don't remove data unrelated to PR

File: src/packs/melee/Suit ripper.yml [hunk 25]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: none
     value: 0
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/melee/Suit ripper.yml:13]: Don't remove data unrelated to PR

File: src/packs/melee/Unarmed attack.yml [hunk 26]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: none
     value: 0
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/melee/Unarmed attack.yml:13]: Don't remove data unrelated to PR

File: src/packs/ranged/Advanced Bow.yml [hunk 27]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: ammo
     value: 1
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/ranged/Advanced Bow.yml:13]: Don't remove data unrelated to PR

File: src/packs/ranged/Combat Rifle.yml [hunk 28]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: ammo
     value: 30
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/ranged/Combat Rifle.yml:13]: Don't remove data unrelated to PR

File: src/packs/ranged/Combat Shotgun.yml [hunk 29]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: ammo
     value: 12
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/ranged/Combat Shotgun.yml:13]: Don't remove data unrelated to PR

File: src/packs/ranged/Conversion Bow.yml [hunk 30]
```diff
@@ -8,9 +8,6 @@ data:
     suppress: false
     type: ammo
     value: 1
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/ranged/Conversion Bow.yml:13]: Don't remove data unrelated to PR

File: src/packs/ranged/Crude Pistol.yml [hunk 31]
```diff
@@ -8,19 +8,13 @@ data:
...
   damage: 1d6
   description: >-
     Crude pistols and muskets represent the rawest and most primitive forms of
     gunpowder weaponry, usually makeshift weapons improvised by criminals or the
     desperate. Reloading a crude pistol or a musket requires two rounds instead
-    of one. Revolvers are quite popular on frontier worlds, as the weapons are
-    extremely reliable and can be repaired and manufactured even by primitive
-    metallurgists. Some revolver variants are specially built to handle
-    atmospheres that would destroy more fragile weapons.
+    of one.
```
Comment (Spice-King) [inline src/packs/ranged/Crude Pistol.yml:17]: Don't remove data unrelated to PR

File: src/packs/ranged/Distortion Cannon.yml [hunk 32]
```diff
@@ -1,16 +1,13 @@
...
   ammo:
     burst: false
     longReload: false
     max: 6
     suppress: false
     type: typeAPower
     value: 6
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/ranged/Distortion Cannon.yml:13]: Don't update or remove data unrelated to PR

File: src/packs/ranged/Laser Pistol.yml [hunk 33]
```diff
@@ -1,16 +1,13 @@
...
   ammo:
     burst: false
     longReload: false
     max: 10
     suppress: false
     type: typeAPower
     value: 10
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/ranged/Laser Pistol.yml:13]: Don't update or remove data unrelated to PR

File: src/packs/ranged/Laser Rifle.yml [hunk 34]
```diff
@@ -1,16 +1,13 @@
...
   ammo:
     burst: true
     longReload: false
     max: 20
     suppress: false
     type: typeAPower
     value: 20
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/ranged/Laser Rifle.yml:13]: Don't update or remove data unrelated to PR

File: src/packs/ranged/Plasma Projector.yml [hunk 35]
```diff
@@ -1,16 +1,13 @@
...
   ammo:
     burst: false
     longReload: false
     max: 6
     suppress: false
     type: typeAPower
     value: 6
-  bundle:
-    amount: 1
-    bundled: false
```
Comment (Spice-King) [inline src/packs/ranged/Plasma Projector.yml:13]: Don't update or remove data unrelated to PR

File: src/packs/ranged/Revolver.yml [hunk 36]
```diff
@@ -8,16 +8,10 @@ data:
...
-    amount: 1
-    bundled: false
   cost: 50
   damage: 1d8
   description: >-
-    Crude pistols and muskets represent the rawest and most primitive forms of
-    gunpowder weaponry, usually makeshift weapons improvised by criminals or the
-    desperate. Reloading a crude pistol or a musket requires two rounds instead
-    of one. Revolvers are quite popular on frontier worlds, as the weapons are
+    Revolvers are quite popular on frontier worlds, as the weapons are
```
Comment (Spice-King) [inline src/packs/ranged/Revolver.yml:14]: Don't remove data unrelated to PR

File: src/packs/ranged/Shear Rifle.yml [hunk 37]
```diff
@@ -1,16 +1,13 @@
...
     burst: true
     longReload: false
     max: 10
     suppress: false
     type: ammo
     value: 10
-  bundle:
-    amount: 1
-    bundled: false
   cost: 600
```
Comment (Spice-King) [inline src/packs/ranged/Shear Rifle.yml:11]: Don't update or remove data unrelated to PR

File: src/packs/ranged/Thermal Pistol.yml [hunk 38]
```diff
@@ -1,16 +1,13 @@
...
     burst: false
     longReload: false
     max: 5
     suppress: false
     type: typeAPower
     value: 5
-  bundle:
-    amount: 1
-    bundled: false
   cost: 300
```
Comment (Spice-King) [inline src/packs/ranged/Thermal Pistol.yml:11]: Don't update or remove data unrelated to PR

File: src/packs/ranged/Thunder Gun.yml [hunk 39]
```diff
@@ -1,16 +1,13 @@
...
     burst: false
     longReload: false
     max: 6
     suppress: false
     type: typeAPower
     value: 6
-  bundle:
-    amount: 1
-    bundled: false
   cost: 1000
```
Comment (Spice-King) [inline src/packs/ranged/Thunder Gun.yml:11]: Don't update or remove data unrelated to PR

File: src/swnr.pcss [hunk 40]
```diff
@@ -28,7 +28,8 @@ a:hover {
   @apply text-shadow-gray-600;
 }
 div.editor .editor-content{
-  min-height: 200px;
+  min-height: 100px;
+  resize: vertical;
```
Comment (Spice-King) [inline src/swnr.pcss:32]: Don't update or remove data unrelated to PR

File: src/system.yml [hunk 41]
```diff
@@ -12,18 +12,6 @@ esmodules:
...
-    system: swnr
-    module: swnr
-    path: packs/foci.db
-    entity: Item
-  - name: ranged
-    label: Ranged Weapons
-    system: swnr
-    module: swnr
-    path: packs/ranged.db
-    entity: Item
```
Comment (Spice-King) [inline src/system.yml:26]: Reordering a list is something that should be done in another PR.

File: src/templates/actors/drone-sheet.html [hunk 42]
```diff
@@ -0,0 +1,278 @@
...
+              {{localize 'swnr.drone.model.pax.name'}}
+            </option>
+            <option value="alecto">
+              {{localize 'swnr.drone.model.alecto.name'}}
+            </option>
+            <option value="other">
+              {{localize 'swnr.drone.model.other.name'}}
+            </option>
+            {{/select}}
+          </select>
```
Comment (Spice-King) [inline src/templates/actors/drone-sheet.html:54]: Again, this is a code smell and locks out custom content. I do not want to treat 3rd party content as a second class citizen.

File: src/templates/actors/fragments/weapon.html [hunk 43]
```diff
@@ -2,6 +2,7 @@
   class="flex gap-1 items-baseline group item weapon"
   data-item-id="{{item._id}}"
 >
+  <img class="w-5 h-5 my-auto align-middle" src="systems/swnr/assets/icons/game-icons.net/item-icons/{{item.type}}.svg"/>
```
Comment (Spice-King) [inline src/templates/actors/fragments/weapon.html:5]: Out of scope for this PR.

File: src/templates/fragments/desc.html [hunk 44]
```diff
@@ -1,10 +1,3 @@
-  <input
-    class="text-lg col-span-3 px-1.5 py-3.5 border border-gray-800 bg-gray-400 bg-opacity-75 placeholder-blue-800 placeholder-opacity-75 rounded-md"
-    name="name"
-    type="text"
-    value="{{item.name}}"
-    placeholder="{{localize 'swnr.sheet.name'}}"
-  />
```
Comment (Spice-King) [inline src/templates/fragments/desc.html:7]: Refactoring this is out of scope for this PR.

**Review by Spice-King — CHANGES_REQUESTED** [review]
This feels like 4-5 different PRs worth of unrelated changes. PRs should be focused on one thing.

**Comment by DroneEnthusiast** [comment]
Thanks for the feedback. Allow me to separate out the changes.

File: package.json (hunk 1 above)
Comment (DroneEnthusiast) [reply package.json:12]: Alright, this won't compile against your typescript definitions without some API changes. Please accept my merge request on gitlab [here](https://gitlab.com/Spice_King/foundry-pc-types/-/merge_requests).

File: src/module/ValidatedDialog.ts (hunk 2 above)
Comment (DroneEnthusiast) [reply src/module/ValidatedDialog.ts:25]: Done as part of #32.

File: src/module/actors/character-sheet.ts (hunk 3 above)
Comment (DroneEnthusiast) [reply src/module/actors/character-sheet.ts:251]: Done as part of #32.

File: src/module/actors/npc.ts (hunk 4 above)
Comment (DroneEnthusiast) [reply src/module/actors/npc.ts:15]: Done as part of #32.

File: src/module/handlebar-helpers.ts (hunk 5 above)
Comment (DroneEnthusiast) [reply src/module/handlebar-helpers.ts:14]: In #35.

File: src/swnr.pcss (hunk 40 above)
Comment (DroneEnthusiast) [reply src/swnr.pcss:32]: Done as part of #37.

File: src/module/items/base-sheet.ts (hunk 6 above)
Comment (DroneEnthusiast) [reply src/module/items/base-sheet.ts:5]: Done as part of #32.

File: src/module/types.d.ts (hunk 7 above)
Comment (DroneEnthusiast) [reply src/module/types.d.ts:81]: Gotcha. Change was made.

File: src/module/actors/character.ts (hunk 8 above)
Comment (DroneEnthusiast) [reply src/module/actors/character.ts:13]: Done as part of #32.

File: src/module/actors/drone-sheet.ts (hunk 9 above)
Comment (DroneEnthusiast) [reply src/module/actors/drone-sheet.ts:217]: Your idea for Ships sounds good to me. I added an item compendium pack for drone models. Drag and drop works and the drone sheet still contains a dropdown which is now populated by index for this pack.

One advantage of this approach is the process is symmetric for all the Actor types: we can universally use `Create New Actor` to make these entities and items to modify them.

File: src/packs/heavy/Anti-Vehicle Laser.yml (hunk 10 above)
Comment (DroneEnthusiast) [reply src/packs/heavy/Anti-Vehicle Laser.yml:13]: You helpfully did this in one of your commits.

File: src/packs/heavy/Demo Charge.yml (hunk 11 above)
Comment (DroneEnthusiast) [reply src/packs/heavy/Demo Charge.yml:13]: You helpfully did this in one of your commits.

File: src/packs/heavy/Heavy Machine Gun.yml (hunk 12 above)
Comment (DroneEnthusiast) [reply src/packs/heavy/Heavy Machine Gun.yml:13]: You helpfully did this in one of your commits.

File: src/packs/heavy/Hydra Array.yml (hunk 13 above)
Comment (DroneEnthusiast) [reply src/packs/heavy/Hydra Array.yml:13]: You helpfully did this in one of your commits.

File: src/packs/heavy/Railgun.yml (hunk 14 above)
Comment (DroneEnthusiast) [reply src/packs/heavy/Railgun.yml:13]: You helpfully did this in one of your commits.

File: src/packs/heavy/Rocket Launcher.yml (hunk 15 above)
Comment (DroneEnthusiast) [reply src/packs/heavy/Rocket Launcher.yml:13]: You helpfully did this in one of your commits.

File: src/packs/heavy/Vortex Cannon.yml (hunk 16 above)
Comment (DroneEnthusiast) [reply src/packs/heavy/Vortex Cannon.yml:13]: You helpfully did this in one of your commits.

File: src/packs/heavy/Wheatcutter Belt.yml (hunk 17 above)
Comment (DroneEnthusiast) [reply src/packs/heavy/Wheatcutter Belt.yml:13]: You helpfully did this in one of your commits.

File: src/packs/melee/Large advanced weapon.yml (hunk 18 above)
Comment (DroneEnthusiast) [reply src/packs/melee/Large advanced weapon.yml:13]: You helpfully did this in one of your commits.

File: src/packs/melee/Large primitive weapon.yml (hunk 19 above)
Comment (DroneEnthusiast) [reply src/packs/melee/Large primitive weapon.yml:13]: You helpfully did this in one of your commits.

File: src/packs/melee/Medium advanced weapon.yml (hunk 20 above)
Comment (DroneEnthusiast) [reply src/packs/melee/Medium advanced weapon.yml:13]: You helpfully did this in one of your commits.

File: src/packs/melee/Medium primitive weapon.yml (hunk 21 above)
Comment (DroneEnthusiast) [reply src/packs/melee/Medium primitive weapon.yml:13]: You helpfully did this in one of your commits.

File: src/packs/melee/Small advanced weapon.yml (hunk 22 above)
Comment (DroneEnthusiast) [reply src/packs/melee/Small advanced weapon.yml:13]: You helpfully did this in one of your commits.

File: src/packs/melee/Small primitive weapon.yml (hunk 23 above)
Comment (DroneEnthusiast) [reply src/packs/melee/Small primitive weapon.yml:13]: You helpfully did this in one of your commits.

File: src/packs/melee/Stun baton.yml (hunk 24 above)
Comment (DroneEnthusiast) [reply src/packs/melee/Stun baton.yml:13]: You helpfully did this in one of your commits.

File: src/packs/melee/Suit ripper.yml (hunk 25 above)
Comment (DroneEnthusiast) [reply src/packs/melee/Suit ripper.yml:13]: You helpfully did this in one of your commits.

File: src/packs/melee/Unarmed attack.yml (hunk 26 above)
Comment (DroneEnthusiast) [reply src/packs/melee/Unarmed attack.yml:13]: You helpfully did this in one of your commits.

File: src/packs/ranged/Advanced Bow.yml (hunk 27 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Advanced Bow.yml:13]: You helpfully did this in one of your commits.

File: src/packs/ranged/Combat Rifle.yml (hunk 28 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Combat Rifle.yml:13]: You helpfully did this in one of your commits.

File: src/packs/ranged/Combat Shotgun.yml (hunk 29 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Combat Shotgun.yml:13]: You helpfully did this in one of your commits.

File: src/packs/ranged/Conversion Bow.yml (hunk 30 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Conversion Bow.yml:13]: You helpfully did this in one of your commits.

File: src/packs/ranged/Crude Pistol.yml (hunk 31 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Crude Pistol.yml:17]: This was handled in #33.

File: src/packs/ranged/Distortion Cannon.yml (hunk 32 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Distortion Cannon.yml:13]: This one was handled in #34.

File: src/packs/ranged/Laser Pistol.yml (hunk 33 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Laser Pistol.yml:13]: This one was handled in #34.

File: src/packs/ranged/Laser Rifle.yml (hunk 34 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Laser Rifle.yml:13]: This one was handled in #34.

File: src/packs/ranged/Plasma Projector.yml (hunk 35 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Plasma Projector.yml:13]: This one was handled in #34.

File: src/packs/ranged/Revolver.yml (hunk 36 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Revolver.yml:14]: This was handled in #33.

File: src/packs/ranged/Shear Rifle.yml (hunk 37 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Shear Rifle.yml:11]: This one was handled in #34.

File: src/packs/ranged/Thermal Pistol.yml (hunk 38 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Thermal Pistol.yml:11]: This one was handled in #34.

File: src/packs/ranged/Thunder Gun.yml (hunk 39 above)
Comment (DroneEnthusiast) [reply src/packs/ranged/Thunder Gun.yml:11]: This one was handled in #34.

File: src/system.yml (hunk 41 above)
Comment (DroneEnthusiast) [reply src/system.yml:26]: Done in #36.

File: src/templates/actors/drone-sheet.html (hunk 42 above)
Comment (DroneEnthusiast) [reply src/templates/actors/drone-sheet.html:54]: That makes sense. Leaving code like this general will also help maintain developers' sanity.

I left an "Other" option in place to allow someone who doesn't want to or cannot modify the compendium to specify a custom model. Let me know what you think.

File: src/templates/actors/fragments/weapon.html (hunk 43 above)
Comment (DroneEnthusiast) [reply src/templates/actors/fragments/weapon.html:5]: Pending in #39.

File: src/templates/fragments/desc.html (hunk 44 above)
Comment (DroneEnthusiast) [reply src/templates/fragments/desc.html:7]: Open @ #38.

**Comment by DroneEnthusiast** [comment]
You can change the base branch of a PR but I think it will make a mess so I am going to close this and create a new PR against master with my updated diff. I will update the PR for the new API when you are done with the other updates for 0.8.x. Let me know if I can help! Otherwise I'll work on something else out of the way when I get free time.
